internal/pgproxy/pgproxy.go

```diff
@@ -0,0 +1,194 @@
+package pgproxy
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/jackc/pgx/v5/pgproto3"
+
+	"github.com/TBD54566975/ftl/internal/log"
+)
+
+type PgProxy struct {
+	// listenAddress is the address to listen on for incoming connections.
+	listenAddress string
+	// connectionStringFn is a function that constructs a new connection string from parameters of the incoming connection.
+	connectionStringFn func(ctx context.Context, params map[string]string) (string, error)
+}
+
+// NewPgProxy creates a new PgProxy.
+func NewPgProxy(address string, connectionFn func(ctx context.Context, params map[string]string) (string, error)) *PgProxy {
```

Comment:
nit: typically for a package with a single type, I'd just use `New()` as the extra name is redundant and stutters: `pgproxy.NewPgProxy`

Reply:
Can you document what "params" is?

Reply:
Or maybe just move the docs from the type fields to the function, as that's public.